src/main/java/com/cleanroommc/modularui/widget/Widget.java

```diff
@@ -326,6 +326,7 @@ public W onUpdateListener(Consumer<W> listener, boolean merge) {
     }
 
     public W setEnabledIf(Predicate<W> condition) {
+        setEnabled(condition.test(getThis()));
         return onUpdateListener(w -> setEnabled(condition.test(w)), true);
     }
```

Comment:
this might cause issues because the widget is not yet initialised. I already call update function in init which should be enough